src/main/scala/com/fortysevendeg/translatebubble/ui/bubbleservice/SharedActivity.scala

```diff
@@ -0,0 +1,41 @@
+package com.fortysevendeg.translatebubble.ui.bubbleservice
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import com.fortysevendeg.macroid.extras.AppContextProvider
+import com.fortysevendeg.translatebubble.modules.ComponentRegistryImpl
+import com.fortysevendeg.translatebubble.modules.clipboard.CopyToClipboardRequest
+import macroid.{AppContext, Contexts}
+
+class SharedActivity
+  extends Activity
+  with Contexts[Activity]
+  with ComponentRegistryImpl
+  with AppContextProvider {
+
+  private val acceptedType = "text/plain"
+
+  override implicit lazy val appContextProvider: AppContext = activityAppContext
+
+  override def onCreate(savedInstanceState: Bundle) = {
+    super.onCreate(savedInstanceState)
+
+    Option(getIntent) map { i =>
+      (Option(i.getAction), Option(i.getType)) match {
+        case (Some(Intent.ACTION_SEND), Some(`acceptedType`)) =>
+          Option(i.getStringExtra(Intent.EXTRA_TEXT)) map handleText
+      }
+    }
```

Comment:
Don't like too much the `for` comprehension here because we should add the remaining logic in the `yield` block or deal with nested *Options* as result of the `for` comprehension. @javipacheco what do you think about this alternative logic?